influunt-api/modules/influunt-core/app/models/ControladorFisico.java

```diff
@@ -258,6 +259,30 @@ public Controlador getControladorSincronizado() {
         return controladorSincronizado;
     }
 
+    public static List<ControladorFisico> getControladorPorUsuario(Usuario usuario) {
+        Area area = null;
+        if (!usuario.isRoot() && !usuario.podeAcessarTodasAreas()) {
+            area = usuario.getArea();
+        }
+
+        return getControladorPorArea(area);
```

Comment:
Tem um problema aqui, se o usuário não for root, não tiver a permissão de acessar todas as áreas e a área dele for `null`, ele vai acessar os dados de todas as áreas. Acho que dentro do `if` acima deveria retornar `null` caso a área do usuário seja `null` também.